src/ResourceManager/DeviceProvisioningServices/Commands.DeviceProvisioningServices/Cmdlet/IotDpsLinkedHubs/UpdateAzureRmIotDeviceProvisioningServiceLinkedHub.cs

```diff
@@ -0,0 +1,141 @@
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.Management.DeviceProvisioningServices.Models;
+    using DPSResources = Microsoft.Azure.Commands.Management.DeviceProvisioningServices.Properties.Resources;
+
+    [Cmdlet(VerbsData.Update, "AzureRmIoTDeviceProvisioningServiceLinkedHub", DefaultParameterSetName = ResourceParameterSet, SupportsShouldProcess = true)]
+    [Alias("Update-AzureRmIoTDpsHub")]
+    [OutputType(typeof(PSIotHubDefinitionDescription))]
+    public class UpdateAzureRmIoTDeviceProvisioningServiceLinkedHub : IotDpsBaseCmdlet
+    {
+        private const string ResourceIdParameterSet = "ResourceIdSet";
+        private const string ResourceParameterSet = "ResourceSet";
+        private const string InputObjectParameterSet = "InputObjectSet";
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = InputObjectParameterSet,
+            ValueFromPipeline = true,
+            HelpMessage = "IoT Device Provisioning Service Linked Hub Object")]
+        [ValidateNotNullOrEmpty]
+        public PSIotHubDefinitionDescription InputObject { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "IoT Device Provisioning Service Resource Id")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceParameterSet,
+            HelpMessage = "Name of the Resource Group")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = ResourceParameterSet,
+            HelpMessage = "Name of the IoT Device Provisioning Service")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            HelpMessage = "Host name of linked IoT Hub")]
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            ParameterSetName = ResourceParameterSet,
+            HelpMessage = "Host name of linked IoT Hub")]
+        [ValidateNotNullOrEmpty]
+        public string LinkedHubName { get; set; }
+
+        [Parameter(
+           Mandatory = false,
+           HelpMessage = "Allocation weight of the IoT Hub")]
+        [ValidateNotNullOrEmpty]
+        public int? AllocationWeight { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Apply allocation policy to the IoT Hub")]
+        [ValidateNotNullOrEmpty]
```

Comment:
@anusapan nit: you can remove this attribute

Reply:
Done